### DISCUSSION TITLE
Status of MPO support for cuTensornet

### DISCUSSION THREAD

**Discussion by Sinestro38** [NVIDIA/cuQuantum#101] [Q&A] [answered]
As I understand right now, using cuTensornet to compute properties of quantum circuits requires the circuits to be fully resolved when ingesting into `CircuitToEinsum`. Is there any way we could access the MPO such that we can have a `CircuitToEinsum` like object but for a certain ansatz and is flexible to different parameterizations?

So could something like
```
mpo_obj = CircuitToEinsum(parameterized_circuit)
sv_1 = mpo_obj.statevector(params_1)
sv_2 = mpo_obj.statevector(params_2)
```
Be possible?

**Comment by yangcal** [comment, marked as answer]
Hello,

Indeed, the circuit must be fully parameterized when creating the `CircuitToEinsum` object. This is because the operands are parsed during `CircuitToEinsum` initialization. Currently there is no way to circumvent this requirement.  

> Is there any way we could access the MPO?
Can you describe what you mean by MPO here? In standard quantum circuit simulations, all gates are represented as rank `2*n` tensor where `n` is the number of qubits involved. Multi-qubit gate operands can be converted to MPO but that's currently beyond the capability of `CircuitToEinsum`.
